dev/benchmarks/test_apps/stocks/lib/main.dart

```diff
@@ -91,19 +89,121 @@ class StocksAppState extends State<StocksApp> {
       debugRepaintRainbowEnabled = _configuration.debugShowRainbow;
       return true;
     }());
-    return MaterialApp(
+    return MaterialApp.router(
       title: 'Stocks',
       theme: theme,
       localizationsDelegates: StockStrings.localizationsDelegates,
       supportedLocales: StockStrings.supportedLocales,
       debugShowMaterialGrid: _configuration.debugShowGrid,
       showPerformanceOverlay: _configuration.showPerformanceOverlay,
       showSemanticsDebugger: _configuration.showSemanticsDebugger,
-      routes: <String, WidgetBuilder>{
-         '/':         (BuildContext context) => StockHome(stocks, _configuration, configurationUpdater),
-         '/settings': (BuildContext context) => StockSettings(_configuration, configurationUpdater),
-      },
+      routePackageParser: StockRoutePackageParser(),
+      routerDelegate: StockRouterDelegate(this),
+    );
+  }
+}
+
+class StockRoutePackageParser extends RoutePackageParser<List<String>> {
+  @override
+  Future<List<String>> parse(RoutePackage routePackage) {
+    String initialRouteName = routePackage.routeName;
+    final List<String> result = <String>[];
+    assert(initialRouteName.startsWith('/') && initialRouteName.isNotEmpty);
+    initialRouteName = initialRouteName.substring(1); // strip leading '/'
+    result.add('/');
+    final List<String> routeParts = initialRouteName.split('/');
+    if (initialRouteName.isNotEmpty) {
+      String routeName = '';
+      for (final String part in routeParts) {
+        routeName += '/$part';
+        result.add(routeName);
+      }
+    }
+    return SynchronousFuture<List<String>>(result);
+  }
+
+  @override
+  Future<RoutePackage> restore(List<String> configuration) {
+    return SynchronousFuture<RoutePackage>(
+      RoutePackage(
+        routeName: configuration.last,
+      )
+    );
+  }
+}
+
+class StockRouterDelegate extends RouterDelegate<List<String>> with PopNavigatorRouterDelegateMixin<List<String>> {
+  StockRouterDelegate(this.stockState) {
+    stockState.addListener(notifyListeners);
+  }
+  StocksAppState stockState;
+
+  final HeroController _heroController = MaterialApp.createMaterialHeroController();
+
+  @override
+  final GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
+
+  @override
+  Future<void> setNewRoutePath(List<String> configuration) {
+    stockState.routesUpdater(configuration);
```

Comment:
If I go to www.mystocksapp.com/foo/bar/blue it will try to will throw an assert in _createPage, which seems undesirable?